tests/foreman/api/test_contentviewversion.py

```diff
@@ -33,960 +33,1013 @@
+    cvv = content_view.version[0].read()
+    assert len(cvv.environment) == 2
+    # Delete the content-view version from selected environments
+    for env in reversed(cvv.environment):
+        content_view.delete_from_environment(env.id)
+    content_view.version[0].delete()
+    # Make sure that content view version is really removed
+    assert len(content_view.read().version) == 0
+
+
+@pytest.mark.upgrade
+@pytest.mark.tier2
+@pytest.mark.skipif((not settings.repos_hosting_url), reason='Missing repos_hosting_url')
+def test_positive_delete_composite_version(module_org):
+    """Create composite content view and publish it. After that try to
+    disassociate content view from 'Library' environment through
+    'delete_from_environment' command and delete content view version from
+    that content view. Add repository to initial content view
+    for better coverage.
+
+    :id: b5bb547e-0174-464c-b974-0254d372cdd6
+
+    :expectedresults: Content version deleted successfully
+
+    :CaseLevel: Integration
+
+    :BZ: 1276479
+    """
+    # Create and publish product/repository
+    product = entities.Product(organization=module_org).create()
+    repo = entities.Repository(product=product, url=FAKE_1_YUM_REPO).create()
+    repo.sync()
+    # Create and publish content views
+    content_view = entities.ContentView(organization=module_org, repository=[repo]).create()
+    content_view.publish()
+    # Create and publish composite content view
+    composite_cv = entities.ContentView(
+        organization=module_org, composite=True, component=[content_view.read().version[0]]
+    ).create()
+    composite_cv.publish()
+    composite_cv = composite_cv.read()
+    # Get published content-view version id
+    assert len(composite_cv.version) == 1
+    cvv = composite_cv.version[0].read()
+    assert len(cvv.environment) == 1
+    # Delete the content-view version from selected env
+    composite_cv.delete_from_environment(cvv.environment[0].id)
+    # Delete the version
+    composite_cv.version[0].delete()
+    # Make sure that content view version is really removed
+    assert len(composite_cv.read().version) == 0
+
+
+@pytest.mark.upgrade
+@pytest.mark.tier3
+@pytest.mark.skipif((not settings.repos_hosting_url), reason='Missing repos_hosting_url')
+def test_positive_delete_with_puppet_content(module_org):
+    """Delete content view version with puppet module content
+
+    :id: cae1164c-6608-4e19-923c-936e75ed807b
+
+    :steps:
+        1. Create a lifecycle environment
+        2. Create a content view
+        3. Add a puppet module to content view
+        4. Publish the content view
+        5. Promote the content view to lifecycle environment
+        6. Remove the content view versions from all lifecycle environments
+        7. Delete the content view version
+
+    :expectedresults: Content view version deleted successfully
+
+    :CaseLevel: Integration
+    """
+    lce_library = (
```

Comment:
its a default lce again not a lib. Use a default_lce fixture instead as suggested above.